handbook/engineering/web/goals.md

```diff
@@ -4,29 +4,24 @@
 
-1. **Make the products that extensions are build on (web app, code host integrations) more consistent and improve discoverability.**
+1. **Make the Sourcegraph web interfaces more consistent and improve discoverability of Sourcegraph features.**
    Our web app has accumulated a lot of design debt over time, which negatively impacts how we can use it as a vehicle to deliver extensions.
-   With areas like the repository page, user settings area (which extensions are configured through), navigation, command palette UI and the extension registry affected, it is hard to provide a good UX around extensions (the extension registry blends into goal (2)).
-   Our code host integrations, which are an implementation of our extension API, are a huge driver of adoption inside companies and multiply the value of extensions by bringing them into code review workflows, but are difficult to discover and setup.
-2. **Bring the extension platform into shape.**
+   With areas like the repository page, user settings area, navigation, settings, command palette UI and overall design affected, we want to provide the best possible UX for using Sourcegraph. 
+2. **Make the browser extension and Sourcegraph extensions a core part of Sourcegraph users' experiences.**
    Our extension platform includes the workflow around creating, installing and using extensions, the API exposed to developers and its documentation.
-   To grow adoption of extensions, these need to be solid, but they are currently lacking on multiple dimensions.
-   Some API areas like code insights are still in prototype phase and are still undocumented (this blends into goal (3)).
-   Our extension host is also currently implemented in a way that makes it difficult for us to maintain, evolve to enable more use cases and to onboard new teammates into this area of the codebase.
-   Combining this with writing a few smaller extensions (part of goal (3)) allows us to dogfood the experience and inform us where the platform is lacking.
-3. **Build out compelling use cases with the extensions platform.**
-   This includes writing more extensions ourselves, but also extending the extension API with more capabilities to enable more use cases.
-   We have a long list of ideas and use cases customers shared with us that we can solve and/or enable customers to solve with extensions.
-   Some of these integrate Sourcegraph with more external services to suit the setups of more customers, others provide completely new value no other product provides.
+   To grow adoption of extensions, these need to be solid and obviously useful. The extensions action bar needs to be clear and scalable, and help users discover our extensions. We want to track anonymized, general usage of extensions to determine which extensions are most successful at adding value for our users. 
+   We have a powerful extension API, but we know of use cases and customer requests that could put adddional endpoints to use. 
```

Comment:
These two lines feel very low-level (mentioning very specific projects, and restricting extension improvements to additional endpoints). When goals are too low-level, they can't be used to inform what projects _not_ mentioned here we should take on. Can we open this up a bit again and talk more about the end goal as opposed to the means to get there? In the previous text I simply used the wording "This includes ..." to give examples while not implying that we'll do that and _only_ that.

Reply:
Updated, and continuing to work on this section tomorrow.